src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp

```diff
@@ -2497,8 +2497,9 @@ namespace OpenMS
           DOMElement* current_cv = current_pep->getOwnerDocument()->createElement(XMLString::transcode("cvParam"));
           current_mod->setAttribute(XMLString::transcode("location"), XMLString::transcode("0"));
           current_mod->setAttribute(XMLString::transcode("monoisotopicMassDelta"), XMLString::transcode(String(mod.getDiffMonoMass()).c_str()));
-          current_mod->setAttribute(XMLString::transcode("residues"), XMLString::transcode(mod.getOrigin().c_str()));
-
+          String origin = mod.getOrigin();
```

Comment:
Why? There's a definition for `string::operator=(char)`. Does it give a warning on your system?